src/zenml/post_execution/pipeline.py

```diff
@@ -89,6 +89,31 @@ def get_run(self, name: str) -> "PipelineRunView":
             )
         return run
 
+    def get_run_for_completed_step(self, step_name: str) -> "PipelineRunView":
+        """This method helps you find out which pipeline run produced
+        the cached artifact of a given step.
+
+        Args:
+            step_name: Name of step at hand
+        Return:
+            None if no run is found that completed the given step,
+             else the original pipeline_run
+        """
+        orig_pipeline_run = None
+
+        for run in reversed(self.runs):
+            try:
+                step = run.get_step(step_name)
+                if step.is_completed:
+                    orig_pipeline_run = run
+                    break
+            except KeyError:
+                pass
+        if not orig_pipeline_run:
+            logger.error("Pipeline Run for cached step could not be found")
```

Comment:
Would it make sense to raise an exception here? If an exception is not raised then I would change the return type to `Optional["PipelineRunView"]`